src/umpire/strategy/DynamicSizePool.hpp

```diff
@@ -432,13 +432,17 @@ class DynamicSizePool : private umpire::strategy::mixins::AlignedAllocation {
     return nb;
   }
 
-  void coalesce()
+  void coalesce(std::size_t suggested_size)
   {
     if (getFreeBlocks() > 1) {
-      std::size_t size_to_coalesce = freeReleasedBlocks();
+      freeReleasedBlocks();
```

Comment:
Maybe I'm missing something, but I'm confused how before this function returned some size_t value and now there is no return value (or it is being ignored). Do you see what I mean?

Reply:
Good question.  `freeReleasedBlocks()` returns the number of bytes that were freed.  The coalesce method is now concerned about what the total size of the pool should be.  So, we call `getActualSize()` after `freeReleasedBlocks()` to determine the new size of the pool.  If that size is smaller than `suggested_size`, we then allocate a block big enough to increase the pools size.

Reply:
Hmm I see. So it is purposely ignored.. could the number of bytes that were freed be useful at all (like in the logging statement that follows)? But ya, that makes sense. Thanks for clarifying!